src/Metadata/GraphQl/Query.php

```diff
@@ -13,66 +13,71 @@
-        ?int $urlGenerationStrategy = null,
-        ?bool $read = null,
-        ?bool $deserialize = null,
-        ?bool $validate = null,
-        ?bool $write = null,
-        ?bool $serialize = null,
-        ?bool $fetchPartial = null,
-        ?bool $forceEager = null,
-        ?int $priority = null,
-        ?string $name = null,
-        $provider = null,
-        $processor = null,
-        array $extraProperties = [],
-
+        protected ?string $shortName = null,
+        protected ?string $class = null,
+        protected ?bool $paginationEnabled = null,
+        protected ?string $paginationType = null,
+        protected ?int $paginationItemsPerPage = null,
+        protected ?int $paginationMaximumItemsPerPage = null,
+        protected ?bool $paginationPartial = null,
+        protected ?bool $paginationClientEnabled = null,
+        protected ?bool $paginationClientItemsPerPage = null,
+        protected ?bool $paginationClientPartial = null,
+        protected ?bool $paginationFetchJoinCollection = null,
+        protected ?bool $paginationUseOutputWalkers = null,
+        protected ?array $paginationViaCursor = null,
+        protected ?array $order = null,
+        protected ?string $description = null,
+        protected ?array $normalizationContext = null,
+        protected ?array $denormalizationContext = null,
+        protected ?string $security = null,
+        protected ?string $securityMessage = null,
+        protected ?string $securityPostDenormalize = null,
+        protected ?string $securityPostDenormalizeMessage = null,
+        protected ?string $securityPostValidation = null,
+        protected ?string $securityPostValidationMessage = null,
+        protected ?string $deprecationReason = null,
+        protected ?array $filters = null,
+        protected ?array $validationContext = null,
+        protected mixed $input = null,
+        protected mixed $output = null,
+        /** @var array|bool|string|null $mercure */
+        protected $mercure = null,
+        /** @var bool|string|null $messenger */
+        protected $messenger = null,
+        protected ?bool $elasticsearch = null,
+        protected ?int $urlGenerationStrategy = null,
+        protected ?bool $read = null,
+        protected ?bool $deserialize = null,
+        protected ?bool $validate = null,
+        protected ?bool $write = null,
+        protected ?bool $serialize = null,
+        protected ?bool $fetchPartial = null,
+        protected ?bool $forceEager = null,
+        protected ?int $priority = null,
+        protected ?string $name = null,
+        /** @var (callable(): mixed)|string|null $provider */
+        protected $provider = null,
+        /** @var (callable(): mixed)|string|null $processor */
+        protected $processor = null,
+        protected array $extraProperties = [],
         protected ?bool $nested = null,
+        protected ?PersistenceMeansInterface $persistenceMeans = null,
     ) {
```

Comment:
with this change are you sure that the parent constructor gets called? Not sure why we didn't used constructor promotion here but there may be a reason. I'd leave that untouched.

Reply:
I'm unsure, I wish we could have something better to share these constructor options but it is a hard problem. unless if we accept that arguments position can change and that specific options are last?

Reply:
final decision on this is to keep the previous code, we will try to find a better way